execution/client/ethclient/rpc/options.go

```diff
@@ -1,41 +0,0 @@
-// SPDX-License-Identifier: BUSL-1.1
-//
-// Copyright (C) 2025, Berachain Foundation. All rights reserved.
-// Use of this software is governed by the Business Source License included
-// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
-//
-// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
-// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
-// VERSIONS OF THE LICENSED WORK.
-//
-// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
-// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
-// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
-//
-// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
-// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
-// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
-// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
-// TITLE.
-
-package rpc
-
-import (
-	"time"
-
-	"github.com/berachain/beacon-kit/primitives/net/jwt"
-)
-
-// WithJWTSecret sets the JWT secret for the RPC client.
-func WithJWTSecret(secret *jwt.Secret) func(rpc *Client) {
```

Comment:
We do use this approach with options in other parts of the codebase, but I personally don't really see much value to it.
We don't usually build structs in multiple ways, i.e. we don't generally have a struct specifying some options and not others.
If I am missing something, LMK